### DISCUSSION TITLE
Returning id of account when using rodauth as a library

### DISCUSSION THREAD

**Discussion by nacamlada** [jeremyevans/rodauth#290] [General]
Hello)
I uses Rodauth as a library and when I creates account i can't retrieve id of account which was created
Example:

```
auth = Rodauth.lib(json: :only) do
   # some code here
end

session = {}
auth.create_account(login: "some_login", password: "very_good_pass", session: session)  # -> after execution it returns nil
```

And I can't identify which id was assigned to this new account
I tried retrieve it from `session`. I found that `session` hash on my application level has object_id: X but in rodauth level it has different (Y) object_id
Seems rodauth does .dup of this `session`. 
Could you please give me advise how i can get id of new created account?

**Comment by janko** [comment]
I was able to get it working with the following Rodauth configuration, which calls an internal method to return the account ID:

```rb
internal_request_configuration do
  after_create_account do
    super()
    _set_internal_request_return_value(account_id)
  end
end
```
```rb
auth.create_account(...) #=> 123
```

I think it would make sense that `create_account` automatically returns the ID of the created account, let's see what Jeremy thinks.

**Comment by jeremyevans** [comment]
I don't want to have custom code for this.  Assuming `create_account` does not error, you can call `account_id_for_login` to get the id.

**Comment by nacamlada** [comment]
Thanks guys, it works)
